packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -29,14 +36,80 @@ function getPlatformFromRequest(req: express.Request | http.IncomingMessage): st
   return String(platform);
 }
 
+async function shouldUseAnonymousManifestAsync(): Promise<boolean> {
+  const currentSession = await UserManager.getSessionAsync();
+  return !currentSession || ConnectionStatus.isOffline();
+}
+
+async function getScopeKeyAndProjectIdForExpoConfigAsync(
+  expoConfig: ExpoConfig
+): Promise<{
+  scopeKey: string;
+  projectId: string | null;
+}> {
+  const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync();
+  if (shouldUseAnonymousManifest) {
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    return {
+      scopeKey: `@${ANONYMOUS_USERNAME}/${expoConfig.slug}-${userAnonymousIdentifier}`,
+      projectId: null,
+    };
+  }
+
+  const user = await UserManager.ensureLoggedInAsync();
+  const username = await UserManager.getCurrentUsernameAsync();
+  const accountName = expoConfig.owner ?? username;
+  const projectName = expoConfig.slug;
+
+  try {
+    const projects = await ApiV2.clientForUser(user).getAsync('projects', {
+      experienceName: `@${accountName}/${projectName}`,
+    });
+    const project = projects[0];
+    return {
+      scopeKey: project.scopeKey,
+      projectId: project.id,
+    };
```

Comment:
I expect in the future we'll change this to read the EAS project ID from a local config file, whether that's expoConfig (generated from app.config.js/app.json) or eas.json.

Reply:
Yeah, that's definitely one option. How we'll support this bridge between eas-cli and expo-cli is still fairly murky.